### PR TITLE
Make the grafana setup work via apiserver proxy.

### PR DESCRIPTION
**Pull request by vishh**
cc @jimmidyson

### CODE REVIEW

File: grafana/Dockerfile [hunk 1]
```diff
@@ -15,6 +15,7 @@ ENV INFLUXDB_HOST influxdb
 ENV INFLUXDB_PORT 8086
 ENV INFLUXDB_USER root
 ENV INFLUXDB_PASSWORD root
+ENV INFLUXDB_EXTERNAL_URL /db
```
Comment (jimmidyson) [inline grafana/Dockerfile]: This should be `/db/`

File: grafana/config.js.tmpl [hunk 2]
```diff
@@ -4,13 +4,13 @@ function (Settings) {
     datasources: {
       'metrics': {
         type: 'influxdb',
-        url: "/db/{{ .Env.INFLUXDB_NAME }}",
+        url: "{{ .Env.INFLUXDB_EXTERNAL_URL }}/{{ .Env.INFLUXDB_NAME }}",
```
Comment (jimmidyson) [inline grafana/config.js.tmpl]: Remove middle slash with `ENV INFLUXDB_EXTERNAL_URL /db/` above

File: grafana/config.js.tmpl [hunk 3]
```diff
@@ -4,13 +4,13 @@ function (Settings) {
...
         username: "{{ .Env.INFLUXDB_USER }}",
         password: "{{ .Env.INFLUXDB_PASSWORD }}"
       },
       'grafana': {
         type: 'influxdb',
-        url: "/db/{{ .Env.GRAFANA_DB_NAME }}",
+        url: "/{{ .Env.INFLUXDB_EXTERNAL_URL }}/{{ .Env.GRAFANA_DB_NAME }}",
```
Comment (jimmidyson) [inline grafana/config.js.tmpl]: Remove middle slash with `ENV INFLUXDB_EXTERNAL_URL /db/` above

File: deploy/kube-config/influxdb-grafana-controller.yaml [hunk 4]
```diff
@@ -19,25 +19,15 @@ desiredState:
               - containerPort: 8083
               - containerPort: 8086
           - name: "grafana"
-            image: "kubernetes/heapster_grafana:v0.5"
+            image: "kubernetes/heapster_influxdb:v0.5"
```
Comment (jimmidyson) [inline deploy/kube-config/influxdb-grafana-controller.yaml]: Shouldn't we separate out grafana & influxdb into separate pods now for easier deployment? Might want to scale influxdb & grafana separately for example.

**Comment by jimmidyson** [comment]
@vishh - i think this we need to eparate out influxdb & grafana into separate RCs & services. Scaling influxdb & grafana separately is definitely something users are going to want to do. Another use case is hosting grafana on publicly accessible servers, but influxdb not being publicly accessible.

File: grafana/Dockerfile (hunk 1 above)
Comment (vishh) [inline grafana/Dockerfile]: Done

File: grafana/config.js.tmpl (hunk 2 above)
Comment (vishh) [inline grafana/config.js.tmpl]: Done

File: grafana/config.js.tmpl (hunk 3 above)
Comment (vishh) [inline grafana/config.js.tmpl]: Done

File: deploy/kube-config/influxdb-grafana-controller.yaml (hunk 4 above)
Comment (vishh) [inline deploy/kube-config/influxdb-grafana-controller.yaml]: Yup. Filed #194.

**Comment by vishh** [comment]
@jimmidyson: Agreed. Thats been a want for quite a while.
Thanks for the review!

**Comment by vishh** [comment]
Ping @vmarmol @rjnagal

**Comment by rjnagal** [comment]
LGTM

**Pull request merged**